sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionStore.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import java.security.Principal;
+import java.util.List;
+
+/**
+ * This interface represents an abstract permission store. A permission store should be a data structure which
+ * supports add, get, and delete operations and can store "String : ArrayListofPermission" pairs or simulate this storage.
+ * The underlying data structure can be options other than a HashMap as long as the implementation
+ * has a way to translate between the required functions and the data structure below.
+ *
+ * For example, an Array-based solution could be implemented by appending the String to the front of every Permission in
+ * the ArrayList and then storing the Permissions in a Permission[]. You would also need to implement the add, get, and delete
+ * methods to support this structure effectively.
+ *
+ * @opensearch.experimental
+ */
+public interface PermissionStore {
```

Comment:
What do you think about removing this interface and pasting 90% of these comments on the implementation?

Reply:
I can do that! I was not sure if we wanted the interface because we would ultimately want to convert the storage into an OpenSearch index. I am not sure how to go about that kind of implementation off the top of my head so was using a more basic HashMap version for now. I figured with the interface, it would straightforward enough to later swap over to an index-based solution. 

Do you think I should still merge the two or is it worth keeping the interface with this in mind?

Reply:
I agree with Peter's point to move the comments to the concrete class itself. The reason being we will only have one class (Permission.java) implementing this interface (PermissionStore) at any time which is what we want. However, that kinda defeats the purpose of having this interface as it cannot be utilized by any other class (unless we are planning to implement a second PermissionStore but I don't see that happening for the foreseeable future 😄)

Reply:
When deep in the 'what are the requirements, inputs, outputs' phase of a code change, adding interfaces is clarifying, and I think it served its purpose well in this PR.  So in this case it was a tool for the process of this change, and not part of the outcomes of this pull request.

Reply:
If you want to keep it in, I don't see harm, but I do like minimal changes when possible - faster to review